components/AppBar/examples/AppBarAnimatedGlitchExample.swift

```diff
@@ -0,0 +1,281 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CoreGraphics
+import UIKit
+
+import MaterialComponents.MaterialAppBar
+import MaterialComponents.MaterialAppBar_ColorThemer
+import MaterialComponents.MaterialAppBar_TypographyThemer
+import MaterialComponents.MaterialColorScheme
+import MaterialComponents.MaterialFlexibleHeader_CanAlwaysExpandToMaximumHeight
+import MaterialComponents.MaterialTabs
+import MaterialComponents.MaterialTypographyScheme
+
+// This example demonstrates issues with flexible header tabs and animations.
+class AppBarAnimatedJumpExample: UIViewController {
+
+  lazy var appBarViewController: MDCAppBarViewController = self.makeAppBar()
+  var colorScheme = MDCSemanticColorScheme()
+  var typographyScheme = MDCTypographyScheme()
+
+  fileprivate let tabs = [
+    SimpleTableViewController(title: "First"),
+    SimpleTableViewController(title: "Second"),
+    SimpleTableViewController(title: "Third"),
+  ]
+  private var currentTab: SimpleTableViewController? = nil
+
+  lazy var tabBar: MDCTabBar = {
+    let tabBar = MDCTabBar()
+
+    tabBar.items = self.tabs.map { $0.tabBarItem }
+
+    tabBar.delegate = self
+    return tabBar
+  }()
+
+  override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
```

Comment:
Does this get called from `-init`? CatalogByConvention doesn't call this initializer directly, it only calls `-init`.

Reply:
Yes, UIViewController.init calls through to `init(nibName:bundle:)`. As the designated initializer it's the preferred override point.